.pre-commit-config.yaml

```
@@ -1,19 +1,10 @@
 repos:
 
-- repo: https://github.com/psf/black
-  rev: 24.10.0
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.7.3
   hooks:
-  - id: black
-
-- repo: https://github.com/pycqa/isort
-  rev: 5.13.2
-  hooks:
-  - id: isort
-
-- repo: https://github.com/PyCQA/flake8
-  rev: 7.1.1
-  hooks:
-  - id: flake8
+    - id: ruff
```

Let's allow Ruff to fix lines when running the linter:

```suggestion
    - id: ruff
      args: [ --fix ]
```

---

@jorgepiloto, this is because the fix flag has been set to **true** in pyproject.toml, so the fix will still occur. Do you prefer removing it from there in favor of this flag?

---

Thanks for pointing this out, @moe-ad. Then, we can merge this.